### PR TITLE
refactor(runtime/permissions): use concrete error types

### PR DESCRIPTION
**Pull request by crowlKats**

### CODE REVIEW

File: cli/args/flags.rs [hunk 1]
```diff
@@ -11053,7 +11052,7 @@ mod tests {
       "script.ts"
     ]);
     assert_eq!(
-      r.unwrap_err().to_string(),
+      dbg!(r.unwrap_err()).to_string(),
```
Comment (bartlomieju) [inline cli/args/flags.rs:11055]: Remove?

**Review by bartlomieju — COMMENTED** [review]
Looks okay to me, I'd like to get second pair of eyes on it before we land. @dsherret could you please take a look?

File: cli/args/flags.rs (hunk 1 above)
Comment (crowlKats) [reply cli/args/flags.rs:11055]:
```suggestion
      r.unwrap_err().to_string(),
```

File: ext/websocket/lib.rs [hunk 2]
```diff
@@ -158,13 +159,10 @@ pub fn op_ws_check_permission_and_cancel_handle<WP>(
...
 {
-  state
-    .borrow_mut::<WP>()
-    .check_net_url(
-      &url::Url::parse(&url).map_err(WebsocketError::Url)?,
-      &api_name,
-    )
-    .map_err(WebsocketError::Permission)?;
+  state.borrow_mut::<WP>().check_net_url(
+    &url::Url::parse(&url).map_err(WebsocketError::Url)?,
```
Comment (dsherret) [inline ext/websocket/lib.rs:163]: Oh, this shouldn't be parsing when we have full permissions (I'll look into this later)

File: runtime/permissions/prompter.rs [hunk 3]
```diff
@@ -117,7 +124,10 @@ fn clear_stdin(
     loop {
       let r = libc::tcflush(STDIN_FD, libc::TCIFLUSH);
       if r != 0 {
-        bail!("clear_stdin failed (tcflush)");
+        return Err(
+          deno_core::error::generic_error("clear_stdin failed (tcflush)")
+            .into(),
```
Comment (dsherret) [inline runtime/permissions/prompter.rs:129]: Might as well just make this an io error then have this return `std::io::Error`?

File: runtime/permissions/prompter.rs [hunk 4]
```diff
@@ -194,18 +205,22 @@ fn clear_stdin(
 
   return Ok(());
 
-  unsafe fn flush_input_buffer(stdin: HANDLE) -> Result<(), AnyError> {
+  unsafe fn flush_input_buffer(
+    stdin: HANDLE,
+  ) -> Result<(), deno_core::error::AnyError> {
```
Comment (dsherret) [inline runtime/permissions/prompter.rs:210]: Recommend returning `std::io::Error` here.

**Review by dsherret — APPROVED** [review]
LGTM other than PrompterError.

**Pull request merged**